src/ristretto/ristretto_keys.rs

```diff
@@ -240,6 +244,19 @@ impl RistrettoPublicKey {
         })
     }
 
+    /// A verifiable group generator using a domain separated hasher
+    pub fn group_generator(label: &str) -> Result<RistrettoPublicKey, HashingError> {
+        // This function requires 512 bytes of data, so let's be opinionated here and use blake2b
+        let hash = DomainSeparatedHasher::<Blake2b, RistrettoGroupGenerator>::new(label).finalize();
+        if hash.as_ref().len() < 64 {
```

Comment:
Is this possible?

Reply:
It's dependent on Blake2b obv. Sarang had the check in. Middle road is using assert_dbg

Reply:
On second thought, it's just a trivial amount of overhead that it doesn't hurt keeping it in, in case blake's impl changes, or someone wants to refactor in future.

Reply:
Ok cool

Reply:
Yeah, including the check is intended to make it clear what the output requirement is, and reduce the likelihood that future changes introduce a weakness.